aten/src/ATen/core/TensorImpl.h

```diff
@@ -1486,17 +1471,16 @@ struct CAFFE2_API TensorImpl : public c10::intrusive_ptr_target {
 //    strong refcount           TODO: pack these into one word
 //    weak refcount
 //    storage pointer
-//    sizes vector (start)
-//    sizes vector (end)
+//    sizes SmallVector
 //    sizes vector (reserved)   TODO: get rid of me
```

Comment:
get rid of this?

Reply:
I don't understand what "sizes vector (reserved)" is. There's a reserved_ flag in TensorImpl that I suspect has something to do with it

Reply:
If you look at how vectors are usually implemented they hold three pointers. One to first valid element, one to the last valid element, and one to the end of the reserved memory block.

Reply:
This list was really meant to show what every byte of this struct compiles to, so you should probably expand the fields of SmallVector

Reply:
Oh I see, thank you for the explanation!